fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -794,19 +794,25 @@ public CommandProcessingResult rejectApplication(final Long loanId, final JsonCo
 
         final AppUser currentUser = getAppUserIfPresent();
 
-        this.loanApplicationTransitionApiJsonValidator.validateRejection(command.json());
+        // validate before loan assembling
+        loanApplicationTransitionValidator.validateRejection(command.json());
 
+        // retrieve/assemble loan
         final Loan loan = retrieveLoanBy(loanId);
 
+        // check client or group
```

Comment:
Please move this into the `loanApplicationTransitionValidator.validateRejection`

Reply:
done